src/vos/vos_query.c

```diff
@@ -232,6 +260,110 @@ query_recx(struct open_query *query, daos_recx_t *recx)
 	return rc;
 }
 
+static int
+query_normal_recx(struct open_query *query, daos_recx_t *recx)
+{
+	struct evt_extent filter_ex;
+	int		  opc;
+
+	memset(recx, 0, sizeof(*recx));
+	/* query visible last recx */
+	opc = EVT_ITER_EMBEDDED | EVT_ITER_VISIBLE | EVT_ITER_SKIP_HOLES;
+	if (query->qt_flags & VOS_GET_MAX)
+		opc |= EVT_ITER_REVERSE;
+
+	filter_ex.ex_lo = 0;
+	filter_ex.ex_hi = ~(uint64_t)0;
+
+	return _query_recx(query, recx, &filter_ex, opc);
+}
+
+static int
+query_ec_normal_recx(struct open_query *query, daos_recx_t *recx,
+		     struct evt_extent *filter_ex, bool skip_pool)
+{
+	struct evt_extent tmp_ex;
+	int		  opc;
+
+	memset(recx, 0, sizeof(*recx));
+	/* query visible last recx */
+	opc = EVT_ITER_EMBEDDED | EVT_ITER_VISIBLE;
+	if (query->qt_flags & VOS_GET_MAX)
+		opc |= EVT_ITER_REVERSE;
+	if (skip_pool)
```

Comment:
skip_punch ?  What is the logic behind using the word "pool" ?

Reply:
ah, it should be skip_hole.